package/samplers/cmamae/sampler.py

```diff
@@ -179,18 +180,27 @@ def after_trial(
         self._validate_param_names(trial.params.keys())
 
         # Store the trial result.
-        self._values_to_tell.append(values)
+        direction0 = study.directions[0]
+        minimize_in_optuna = study.StudyDirection.MINIMIZE == direction0
+        assert values is not None, "MyPy redefinition."
+        modified_values = list([float(v) for v in values])
+        if minimize_in_optuna:
+            # The direction of the first objective (pyribs maximizes).
+            modified_values[0] = -values[0]
+        self._values_to_tell.append(modified_values)
+        self._stored_trial_numbers.append(trial.number)
 
         # If we have not retrieved the whole batch of solutions, then we should
         # not tell() the results to the scheduler yet.
         if len(self._values_to_tell) != self._batch_size:
             return
 
         # Tell the batch results to external sampler once the batch is ready.
-        values = np.asarray(self._values_to_tell)
+        values_to_tell = np.asarray(self._values_to_tell)[np.argsort(self._stored_trial_numbers)]
```

Comment:
As `values` is passed from the argument, which is actually `Optional[Sequence[float]]`, we need to use another variable name here to avoid the MyPy error.
MyPy was complaining because if `values` is `None`, which can be guessed from `Optional`, we cannot use slicing below.